### DISCUSSION TITLE
Why alertmanagers can not share volumes

### DISCUSSION THREAD

**Discussion by oalimerko** [prometheus-operator/prometheus-operator#5283] [Q&A]
i am running an AKS cluster and i have deployed Prometheus and Alertmanager via deployment resources in k8s and they also are controlled by replicaset.The issue is that sometimes the restart of Alertmanger get stuck.It is related to accessMode of PVC.During restart,k8s will start the new pod in a different node from the currently node where the running pod is assigned,depending on resource utilization on the node.In simple words it means,same PVC is accessed from 2 different pods assigned to different nodes.This is not allowed because in the config of PVC i am using `accessMode ReadWriteOnce`.Looking this [comment](https://github.com/prometheus-operator/prometheus-operator/issues/978#issuecomment-364863367) in github for `prometheus operator` seems to be by design that option `accessMode ReadWriteMany` is not allowed.

So my questions, why such design and what could happen if i change accessMode to  ReadWriteMany?Any practical experience?

**Comment by simonpasquier** [comment]
Do you mean that you've got 2 alertmanager pods with the same identity running at the same time on different nodes? Can you share more details (Alertmanager manifest, kubectl output)?

**Reply by oalimerko** [reply]
suppose you have an AKS with 2 nodes nodeA and nodeB.On nodeA is running/assigned the current alertmanager pod AM1.Now imagine your are going to perform a rolling restart of AM1.Depending on resource utilization on the nodes ,k8s sometimes assigns the new created pod AM2 to nodeB.So it means
AM1 running on nodeA
AM2 running on nodeB(this is how rolling restart works.It creates a new pod and will terminate the old pod until the new one i up & running)
In this scenario,AM2 will never start because his PVC is locked by AM1,PVC can be accessed only by one pod therefore was my question why 2 AM pods can not share same PVC?

In my AKS i have deployed some custom applications developed by my team and they share the PVC between pods using `accessMode ReadWriteOnce` so why AM and prometheus are designed to not share the PVC?

**Comment by simonpasquier** [comment]
Because both pods writing at the same time to the same directory would cause data corruption. If you want a HA Alertmanager setup, you need to set `spec.replicas: 2`.

**Reply by oalimerko** [reply]
if i set `spec.replicas: 2` would not both pods writing at the same time to the same directory?Would this not cause data corruption?

**Comment by simonpasquier** [comment]
you'd have 2 PVCs (one for each pod).
